Generic interface without type argument crashes C compiler

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 20.04.4 LTS (WSL)
Processor: 16 cpus, 64bit, little endian, AMD Ryzen 7 2700X Eight-Core Processor
CC version: cc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

getwd: /mnt/a/source/test v
vmodules: /home/zacaj/.vmodules
vroot: /mnt/a/source/test v/v
vexe: /mnt/a/source/test v/v/v
vexe mtime: 2022-08-03 20:48:29
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 372857c.161ac34

Git version: git version 2.25.1
Git vroot status: weekly.2022.31-14-g161ac343-dirty
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 827f7452

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg test.v`
```v

interface Output<T> {
  val T
  name string
}

struct Coil {
  pub mut: val bool
  pub: name string [required]
}
struct Light {
  pub mut: val string
  pub: name string [required]
}

fn main() {
  mut outs := []Output{}
  outs << Coil { name: 'outhole' }
  outs << Light { name: 'shoot again' }
}

```

**What did you expect to see?**

'outs' definition to either error because I didn't supply a <type> for Output, or to allow me to make an array of mixed Output<T>


**What did you see instead?**
```
==================
/tmp/v_1000/test.16856797593442580990.tmp.c:12239: error: 'main__Output_T_T' undeclared
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/containers.v
+++ b/vlib/v/checker/containers.v
@@ -24,6 +24,30 @@ pub fn (mut c Checker) array_init(mut node ast.ArrayInit) ast.Type {
 							node.elem_type_pos)
 					}
 				}
+			} else if elem_sym.kind == .interface_ {
+				elem_info := elem_sym.info as ast.Interface
+				if elem_info.generic_types.len > 0 && elem_info.concrete_types.len == 0
+					&& !node.elem_type.has_flag(.generic) {
+					if c.table.cur_concrete_types.len == 0 {
+						c.error('generic interface must specify type parameter, e.g. Foo<int>',
+							node.elem_type_pos)
+					} else {
+						c.error('generic interface must specify type parameter, e.g. Foo<T>',
+							node.elem_type_pos)
+					}
+				}
+			} else if elem_sym.kind == .sum_type {
+				elem_info := elem_sym.info as ast.SumType
+				if elem_info.generic_types.len > 0 && elem_info.concrete_types.len == 0
+					&& !node.elem_type.has_flag(.generic) {
+					if c.table.cur_concrete_types.len == 0 {
+						c.error('generic sumtype must specify type parameter, e.g. Foo<int>',
+							node.elem_type_pos)
+					} else {
+						c.error('generic sumtype must specify type parameter, e.g. Foo<T>',
+							node.elem_type_pos)
+					}
+				}
 			}
 		}
 		if node.exprs.len == 0 {

--- a/vlib/v/checker/tests/generic_interface_missing_type_names_err.out
+++ b/vlib/v/checker/tests/generic_interface_missing_type_names_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/generic_interface_missing_type_names_err.vv:16:17: error: generic interface must specify type parameter, e.g. Foo<int>
+   14 |
+   15 | fn main() {
+   16 |   mut outs := []Output{}
+      |                 ~~~~~~
+   17 |   outs << Coil { name: 'outhole' }
+   18 |   outs << Light { name: 'shoot again' }

--- a/vlib/v/checker/tests/generic_interface_missing_type_names_err.vv
+++ b/vlib/v/checker/tests/generic_interface_missing_type_names_err.vv
@@ -1,0 +1,19 @@
+interface Output<T> {
+  val T
+  name string
+}
+
+struct Coil {
+  pub mut: val int
+  pub: name string [required]
+}
+struct Light {
+  pub mut: val int
+  pub: name string [required]
+}
+
+fn main() {
+  mut outs := []Output{}
+  outs << Coil { name: 'outhole' }
+  outs << Light { name: 'shoot again' }
+}

